### PR TITLE
Moving SafeDeleteContext and SafeFreeCredentials to common location

### PR DESCRIPTION
**Pull request by kapilash**
SafeFreeCredentials and SafeDeleteContext are used across two native libraries - libssl and System.Net.Security.Native.
Moving these two classes a common location.

### CODE REVIEW

**Comment by kapilash** [comment]
This is in response to a [PR comment received](https://github.com/dotnet/corefx/pull/6469#discussion_r56659315)

**Comment by kapilash** [comment]
cc: @stephentoub

**Comment by stephentoub** [comment]
cc: @bartonjs

File: src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs [hunk 1]
```diff
@@ -213,43 +198,6 @@ protected override bool ReleaseHandle()
...
-        protected override bool ReleaseHandle()
-        {
-            Debug.Assert((null != _credential), "Null credential in SafeDeleteContext");
-            _credential.DangerousRelease();
-            _credential = null;
-            return true;
-        }
-    }
-
     internal sealed class SafeDeleteSslContext : SafeDeleteContext
```
Comment (bartonjs) [inline src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs]: Everything left in this file (for example, this class) should be also moved to new files so we can delete the whole libssl directory

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Debug Build and Test please

File: src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs (hunk 1 above)
Comment (kapilash) [inline src/Common/src/Interop/Unix/libssl/SecuritySafeHandles.cs]: Split the file into different classes. I moved the classes which are under ```System.Net.Security``` namespace to ```$(COMMONPATH)\System\Net\Security\Unix``` and those in  default namespace (Interop.*) to ```$(COMMONPATH)\Interop\Unix\System.Security.Cryptography.Native``` 
Please let me know if you have any other location in mind

**Comment by kapilash** [comment]
@dotnet-bot test outerloop rhel7.2 release

**Comment by kapilash** [comment]
Outerloop test failure is due to 

        08:38:16 Unhandled Exception: System.InvalidOperationException: This operation is only allowed using a successfully authenticated context.
        08:38:16    at System.Net.Security.SslState.CheckThrow(Boolean authSucessCheck)
        08:38:16    at System.Net.Security.SslState.get_SecureStream()
        08:38:16    at System.Net.Security.SslStream.Read(Byte[] buffer, Int32 offset, Int32 count)
        08:38:16    at System.IO.Stream.<>c.<BeginReadInternal>b__38_0(Object <arg>)
        08:38:16    at System.Threading.Tasks.Task`1.InnerInvoke()
        08:38:16    at System.Threading.Tasks.Task.Execute()
        08:38:16 --- End of stack trace from previous location where exception was thrown ---
        08:38:16    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
        08:38:16    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
        08:38:16    at System.IO.Stream.EndRead(IAsyncResult asyncResult)
        08:38:16    at System.Net.Security.Tests.DummyTcpServer.OnReceive(IAsyncResult result)
        08:38:16    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
        08:38:16    at System.IO.Stream.ReadWriteTask.System.Threading.Tasks.ITaskCompletionAction.Invoke(Task completingTask)
        08:38:16    at System.Threading.Tasks.Task.FinishContinuations()
        08:38:16    at System.Threading.Tasks.Task.Finish(Boolean bUserDelegateExecuted)
        08:38:16    at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
        08:38:16    at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)
        08:38:16    at System.Threading.ThreadPoolWorkQueue.Dispatch()
        08:38:30 ./RunTests.sh: line 198: 33435 Aborted                 (core dumped) ./corerun xunit.console.netcore.exe System.Net.Security.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=IgnoreForCI -notrait category=failing -notrait category=nonlinuxtests


I think this is same as #8472 (which should get fixed with new packages)

File: src/Common/src/System/Net/Security/Unix/SafeFreeCertContext.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+//using System.Security.Authentication;
+//using System.Security.Authentication.ExtendedProtection;
```
Comment (bartonjs) [inline src/Common/src/System/Net/Security/Unix/SafeFreeCertContext.cs]: These seem like they could be deleted...

**Comment by kapilash** [comment]
Both the inner loop failures are in System.Runtime.Loader.Tests. Not yet sure why they fail only on CentOS
    
                            System.Runtime.Loader.Tests.AssemblyLoadContextTest.LoadFromAssemblyName_ValidTrustedPlatformAssembly [FAIL]
          06:51:24       Assert.Throws() Failure
          06:51:24       Expected: typeof(System.IO.FileLoadException)
          06:51:24       Actual:   (No exception was thrown)
          06:51:24       Stack Trace:
          06:51:24             at  System.Runtime.Loader.Tests.AssemblyLoadContextTest.LoadFromAssemblyName_ValidTrustedPlatformAssembly()

**Comment by kapilash** [comment]
@dotnet-bot test outerloop rhel7.2 release

**Comment by kapilash** [comment]
@dotnet-bot test innerloop  CentOS7.1 debug release please

File: src/Common/src/System/Net/Security/Unix/SafeFreeCertContext.cs (hunk 2 above)
Comment (kapilash) [inline src/Common/src/System/Net/Security/Unix/SafeFreeCertContext.cs]: thank you. I missed these. Removed them now

**Comment by kapilash** [comment]
@dotnet-bot test this please

**Comment by bartonjs** [comment]
Looks a lot better.  Thanks, Hemanth.

**Pull request merged**